### PR TITLE
Feature/license from url

### PR DESCRIPTION
**Pull request by timovandeput**
Automatically converts URLs in declared licenses to the licence(s) in the referenced document.

- Maven harvester now prefers reporting license URLs (because they are more concise)
- Rewrite of the license uses the same trust level as the original value
- Licenses scanned from a URL are sorted on score
- License names from ScanCode that include "LicenseRef" in their id are deprioritized
- Downloader now supports redirects (which is not default Java behavior)

### CODE REVIEW

File: src/main/java/com/philips/research/bombase/core/scanner/domain/ScannerInteractor.java [hunk 1]
```diff
@@ -5,25 +5,49 @@
...
     public ScanResult scan(Path directory) {
         return scanner.scan(directory);
     }
+
+    @Override
+    public List<String> scanLicenses(URI uri) {
+        final var licenses = downloader.download(uri, scanner::scan)
+                .getLicenses().stream()
+                .sorted((l, r) -> Integer.compare(r.getScore(), l.getScore()))
+                .peek(lic -> System.out.println(lic.getExpression() + " [" + lic.getScore() + "%]"))
```
Comment (Brend-Smits) [inline src/main/java/com/philips/research/bombase/core/scanner/domain/ScannerInteractor.java:47]: Is this line necessary? We already log on line 50 with LOG.info right?
```suggestion
```

**Review by Brend-Smits — APPROVED** [review]
Looks good, I see that there are still a few open TODO's. Should TODO's be merged into Develop?

**Review by AnnieJovitha — APPROVED** [review]

File: src/main/java/com/philips/research/bombase/core/downloader/domain/AnonymousVcsHandler.java [hunk 2]
```diff
@@ -54,14 +60,14 @@ private File target(Path directory, URI location) {
...
-            final var url = fromUri.toURL();
-            ReadableByteChannel inChannel = Channels.newChannel(url.openStream());
+            ReadableByteChannel inChannel = Channels.newChannel(openStream(fromUri));
             FileChannel outChannel = out.getChannel();
             outChannel.transferFrom(inChannel, 0, Long.MAX_VALUE);
             inChannel.close();
         } catch (IOException e) {
             throw new DownloadException("File download failed from " + fromUri, e);
         }
+        System.out.println("File has size: " + target.length());
```
Comment (JeroenKnoops) [inline src/main/java/com/philips/research/bombase/core/downloader/domain/AnonymousVcsHandler.java:70]:
```suggestion
        LOG.info("File has size: {} ", target.length());
```
or remove

**Review by JeroenKnoops — CHANGES_REQUESTED** [review]
Small suggestion regarding a log message.

File: src/main/java/com/philips/research/bombase/core/scanner/domain/ScannerInteractor.java (hunk 1 above)
Comment (timovandeput) [reply src/main/java/com/philips/research/bombase/core/scanner/domain/ScannerInteractor.java:47]: Oops; this is debugging code I forgot to remove; good catch!

**Pull request merged**